Implements a pattern checker directive for use in the Add Data UI

Bargs opened the pull request:
Related to #5974 

Provides a re-useable directive for checking the document count of a given index pattern. Intended for use on the final step of add data wizards so that the user can verify that their filebeat/topbeat/foobeat install is working.

<img width="473" alt="screen shot 2016-03-09 at 7 55 25 pm" src="https://cloud.githubusercontent.com/assets/6239176/13655919/f52306d8-e630-11e5-9248-b1a8f00c2113.png">

rashidkpc commented on src/ui/public/pattern_checker/pattern_checker.js:
```diff
@@ -0,0 +1,56 @@
+      };
+
+      this.validateInstall = () => {
+        es.count({
```
We want to eventually get the es client out of the frontend.
Can we make this a REST endpoint?

Bargs replied:
@rashidkpc I asked myself the same question, I didn't know if it made sense for something so simple but if we really want to eliminate the ES client completely then I guess it does.

Any opinions on what the endpoint for such an API should be? I could see any of the following being viable options:

`localhost:5601/api/kibana/{indexPatternId}/_count`
(mirrors ES syntax)

`localhost:5601/api/kibana/ingest/{indexPatternId}/_count`
(with the ingest API we've sort of established the ingest config as a higher level concept than an index pattern, so maybe it belongs under ingest?)

`localhost:5601/api/kibana/index_patterns/{indexPatternId}/_count`
(this one feels weird but I include it because it's a way to namespace the API without re-using ingest)

Something else?

rashidkpc replied:
I like the first one, i don't think it belongs under ingest

rashidkpc commented on src/plugins/elasticsearch/index.js:
```diff
@@ -44,6 +44,7 @@ module.exports = function ({ Plugin }) {
       createProxy(server, 'POST', '/{index}/_field_stats');
       createProxy(server, 'POST', '/_msearch');
       createProxy(server, 'POST', '/_search/scroll');
+      createProxy(server, 'POST', '/{index}/_count');
```
I'd rather not add anything new to the proxy, we're working to get rid of it

rashidkpc commented on src/ui/public/pattern_checker/pattern_checker.js:
```diff
@@ -0,0 +1,56 @@
+          }
+        )
+        .then(() => {
+          validationTimeout = $timeout(this.validateInstall, 5000);
```
You need to cancel this when the scope is destroyed, otherwise it will go on querying forever

rashidkpc commented:
I'd get rid of both the "Validating" and "Done" buttons here and make them one big disabled button with the text "Waiting for data ..." and just go ahead and start the looper immediately. Then when you see data change the button to green with a big "Success!" and the text "click to above to continue". Or something like that.

Under the big "Waiting for data" button you can stick a small link that says "Skip this step" for folks who don't need your help watching for documents.

Bargs commented:
Once https://github.com/elastic/kibana/pull/6534 is merged I'll rebase this and then it'll be ready for another look.

Bargs commented:
I've addressed all the feedback, but https://github.com/elastic/kibana/pull/6553 needs to be merged before this branch can be fully tested.

Instead of creating a disabled button that says "waiting for data", I just removed the button entirely and added an extra step to the instructions explaining that we're polling the index pattern and that they can skip the step by simply clicking the existing "Done" button if they want to.

Bargs commented:
@rashidkpc ready for another look.

rashidkpc commented:
Needs test, but otherwise GTG

Bargs commented:
jenkins, test it

Bargs commented:
Tests added and build is finally green after merging master a few times, ready for one more look @rashidkpc

rashidkpc commented:
LGTM
